ui/src/components/integrationFormFields/integrationFormFields.jsx

```diff
@@ -14,7 +15,7 @@ export const IntegrationFormFields = (props) => {
     updateMetaData({
       [SECRET_FIELDS_KEY]: ['apiToken'],
     });
-  }, []);
+  }, [SECRET_FIELDS_KEY, initialData, initialize, updateMetaData]);
```

Comment:
Pls remove the deps.
It is intended to be executed only on component mount phase.